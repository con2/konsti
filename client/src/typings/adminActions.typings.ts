import { Game } from 'typings/game.typings';

export const SUBMIT_UPDATE_HIDDEN = 'SUBMIT_UPDATE_HIDDEN';
export const SUBMIT_GET_SETTINGS = 'SUBMIT_GET_SETTINGS';
export const SUBMIT_SELECT_SIGNUP_TIME = 'SUBMIT_SELECT_SIGNUP_TIME';
export const SUBMIT_SET_TEST_TIME = 'SUBMIT_SET_TEST_TIME';
export const SUBMIT_TOGGLE_APP_OPEN = 'SUBMIT_TOGGLE_APP_OPEN';
export const SUBMIT_RESPONSE_MESSAGE = 'SUBMIT_RESPONSE_MESSAGE';

export interface SubmitUpdateHiddenAsync {
  type: typeof SUBMIT_UPDATE_HIDDEN;
  hiddenGames: readonly Game[];
}

export interface SubmitGetSettingsAsync {
  type: typeof SUBMIT_GET_SETTINGS;
  hiddenGames: readonly Game[];
  signupTime: string;
  appOpen: boolean;
}

export interface SubmitSignupTimeAsync {
  type: typeof SUBMIT_SELECT_SIGNUP_TIME;
  signupTime: string;
}

export interface SubmitSetTestTime {
  type: typeof SUBMIT_SET_TEST_TIME;
  testTime: string;
}

export interface SubmitToggleAppOpenAsync {
  type: typeof SUBMIT_TOGGLE_APP_OPEN;
  appOpen: boolean;
}

export interface SubmitResponseMessageAsync {
  type: typeof SUBMIT_RESPONSE_MESSAGE;
  responseMessage: string;
}

export type AdminActionTypes =
  | SubmitUpdateHiddenAsync
  | SubmitGetSettingsAsync
  | SubmitSignupTimeAsync
  | SubmitSetTestTime
  | SubmitToggleAppOpenAsync
  | SubmitResponseMessageAsync;
