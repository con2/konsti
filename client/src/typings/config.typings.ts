export interface Config {
  appName: string;
  SIGNUP_END_TIME: number;
  SIGNUP_OPEN_TIME: number;
  MESSAGE_DELAY: number;
  CONVENTION_NAME: ConventionName;
  CONVENTION_YEAR: string;
  CONVENTION_START_TIME: string;
  DAY_START_TIME: number;
  apiServerURL: string;
  useTestTime: boolean;
  reduxTrace: boolean;
  loadedSettings: string;
  enableAxe: boolean;
  enableWhyDidYouRender: boolean;
  dataUpdateInterval: number;
  noSignupGames: string[];
  revolvingDoorEnabled: boolean;
  tagFilteringEnabled: boolean;
  simpleDetails: boolean;
}

type ConventionName = 'Ropecon' | 'Tracon Hitpoint';
