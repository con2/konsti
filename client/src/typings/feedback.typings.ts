export interface Feedback {
  feedback: string;
  gameId: string;
}

export interface PostFeedbackResponse {
  message: string;
  status: 'success';
}
