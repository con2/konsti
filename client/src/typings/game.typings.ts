export interface Game {
  gameId: string;
  title: string;
  description: string;
  location: string;
  startTime: string;
  mins: number;
  tags: readonly string[];
  genres: readonly string[];
  styles: readonly string[];
  language: string;
  endTime: string;
  people: string;
  minAttendance: number;
  maxAttendance: number;
  gameSystem: string;
  englishOk: boolean;
  childrenFriendly: boolean;
  ageRestricted: boolean;
  beginnerFriendly: boolean;
  intendedForExperiencedParticipants: boolean;
  shortDescription: string;
  revolvingDoor: boolean;
  popularity: number;
  programType: string;
}

export interface DnDUpdatedPositions {
  availableGames?: readonly Game[];
  selectedGames?: readonly Game[];
}

export interface DnDMove {
  index: number;
  droppableId: string;
}

export interface PostGamesUpdateResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface GetGamesResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface PostHiddenResponse {
  hiddenGames: Game[];
  message: string;
  status: 'success';
}
