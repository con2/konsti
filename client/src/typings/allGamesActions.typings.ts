import { Game } from 'typings/game.typings';

export const SUBMIT_GET_GAMES = 'SUBMIT_GET_GAMES';

export interface SubmitGetGamesAsync {
  type: typeof SUBMIT_GET_GAMES;
  games: readonly Game[];
}

export type AllGamesActionTypes = SubmitGetGamesAsync;
