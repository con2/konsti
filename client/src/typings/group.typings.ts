import { Signup } from 'typings/user.typings';

export interface GroupMember {
  enteredGames: readonly Signup[];
  groupCode: string;
  serial: string;
  signedGames: readonly Signup[];
  username: string;
}

export interface GroupData {
  groupCode: string;
  leader: boolean;
  ownSerial: string;
  username: string;
  leaveGroup?: boolean;
  closeGroup?: boolean;
}

export interface PostGroupResponse {
  groupCode: string;
  message: string;
  status: 'success';
}

export interface GetGroupResponse {
  message: string;
  results: GroupMember[];
  status: 'success';
}
