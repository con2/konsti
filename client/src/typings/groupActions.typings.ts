import { GroupMember } from 'typings/group.typings';

export const SUBMIT_UPDATE_GROUP_CODE = 'SUBMIT_UPDATE_GROUP_CODE';
export const SUBMIT_LEAVE_GROUP = 'SUBMIT_LEAVE_GROUP';
export const SUBMIT_UPDATE_GROUP_MEMBERS = 'SUBMIT_UPDATE_GROUP_MEMBERS';

export interface SubmitUpdateGroupCodeAsync {
  type: typeof SUBMIT_UPDATE_GROUP_CODE;
  groupCode: string;
}

export interface SubmitGetGroupAsync {
  type: typeof SUBMIT_UPDATE_GROUP_MEMBERS;
  groupMembers: readonly GroupMember[];
}

export interface SubmitLeaveGroupAsync {
  type: typeof SUBMIT_LEAVE_GROUP;
  groupCode: string;
}

export type GroupActionTypes =
  | SubmitUpdateGroupCodeAsync
  | SubmitGetGroupAsync
  | SubmitLeaveGroupAsync;
