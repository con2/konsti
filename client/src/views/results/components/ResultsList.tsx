import React, { FC, ReactElement, ChangeEvent } from 'react';
import { useTranslation } from 'react-i18next';
import { ResultsByGameTitle } from './ResultsByGameTitle';
import { ResultsByUsername } from './ResultsByUsername';
import { Result } from 'typings/result.typings';
import styled from 'styled-components';

export interface Props {
  results: readonly Result[];
}

export const ResultsList: FC<Props> = (props: Props): ReactElement => {
  const { results } = props;
  const { t } = useTranslation();
  const [sortedBy, setSortedBy] = React.useState<string>('');
  const [searchTerm, setSearchTerm] = React.useState<string>('');
  const [searchResults, setSearchResults] = React.useState<readonly Result[]>(
    []
  );
  React.useEffect(() => {
    setSortedBy('username');
  }, []);

  React.useEffect(() => {
    setSearchResults(
      results.filter((result) => {
        return (
          result.username.toLowerCase().includes(searchTerm.toLowerCase()) ||
          result.enteredGame.gameDetails.title
            .toLowerCase()
            .includes(searchTerm.toLowerCase())
        );
      })
    );
  }, [searchTerm, results]);

  const buttons = ['username', 'gameTitle'];

  const handleSearchFieldChange = (e: ChangeEvent<HTMLInputElement>): void => {
    setSearchTerm(e.target.value);
  };

  return (
    <div className='results-list'>
      <div className='results-button-row'>
        <span>{t('sortBy')} </span>
        {buttons.map((name) => {
          return (
            <button
              disabled={sortedBy === name}
              value={name}
              onClick={() => setSortedBy(name)}
              key={name}
            >
              {t(name)}
            </button>
          );
        })}
        <FindField>
          <span>{t('find')} </span>
          <span>
            <Input
              type='text'
              value={searchTerm}
              onChange={handleSearchFieldChange}
            />
          </span>
        </FindField>
      </div>
      {sortedBy === 'username' && (
        <ResultsByUsername results={searchResults ?? results} />
      )}
      {sortedBy === 'gameTitle' && (
        <ResultsByGameTitle results={searchResults ?? results} />
      )}
    </div>
  );
};

const FindField = styled.div`
  min-height: 25px;
  max-height: 25px;
  margin: 10px auto;
`;

const Input = styled.input`
  &:active,
  &:focus {
    min-height: 25px;
    max-height: 25px;
  }
`;
