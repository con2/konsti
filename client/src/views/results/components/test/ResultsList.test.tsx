import React from 'react';
import { shallow } from 'enzyme';
import { ResultsList, Props } from '../ResultsList';
import { Result } from 'typings/result.typings';

const results: Result[] = [];

describe('ResultsList', () => {
  it('should render correctly', () => {
    const props: Props = { results };
    const component = shallow(<ResultsList {...props} />);
    expect(component).toMatchSnapshot();
  });
});
