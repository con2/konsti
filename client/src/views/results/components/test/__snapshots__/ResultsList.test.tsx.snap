// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList should render correctly 1`] = `
<div
  className="results-list"
>
  <div
    className="results-button-row"
  >
    <span>
      sortBy
       
    </span>
    <button
      disabled={false}
      key="username"
      onClick={[Function]}
      value="username"
    >
      username
    </button>
    <button
      disabled={false}
      key="gameTitle"
      onClick={[Function]}
      value="gameTitle"
    >
      gameTitle
    </button>
    <ResultsList__FindField>
      <span>
        find
         
      </span>
      <span>
        <ResultsList__Input
          onChange={[Function]}
          type="text"
          value=""
        />
      </span>
    </ResultsList__FindField>
  </div>
</div>
`;
